docs/guides/postgres/clustering/streaming_replication.md

```diff
@@ -0,0 +1,327 @@
+
+Here, both `ha-postgres-1` and `ha-postgres-2` are streaming asynchronously from *primary* server.
+
+#### Automatic failover
+
+If *primary* server fails, another *standby* server will take over and serve as *primary*.
+
+Delete Pod `ha-postgres-0` to see the failover behavior.
+
+```console
+$ kubectl delete pod -n demo ha-postgres-0
+```
+
+```console
+$ kubectl get pods -n demo --selector="kubedb.com/name=ha-postgres" --show-labels
+NAME            READY     STATUS    RESTARTS   AGE       LABELS
+ha-postgres-0   1/1       Running   0          9s        kubedb.com/role=replica,kubedb.com/kind=Postgres,kubedb.com/name=ha-postgres,statefulset.kubernetes.io/pod-name=ha-postgres-0,controller-revision-hash=ha-postgres-69c84579bb
+ha-postgres-1   1/1       Running   0          6m        kubedb.com/role=primary,kubedb.com/kind=Postgres,kubedb.com/name=ha-postgres,statefulset.kubernetes.io/pod-name=ha-postgres-1,controller-revision-hash=ha-postgres-69c84579bb
+ha-postgres-2   1/1       Running   0          6m        kubedb.com/role=replica,kubedb.com/kind=Postgres,kubedb.com/name=ha-postgres,statefulset.kubernetes.io/pod-name=ha-postgres-2,controller-revision-hash=ha-postgres-69c84579bb
+```
+
+Here,
+
+- Pod `ha-postgres-1` is now serving as *primary* server
+- Pod `ha-postgres-0` and `ha-postgres-2` both are serving as *standby* server
+
+And result from `pg_stat_replication`
+
+```console
+postgres=# select * from pg_stat_replication;
+```
+
+ pid | usesysid | usename  | application_name | client_addr | client_port |         backend_start         |   state   | sent_location | write_location | flush_location | replay_location | sync_priority | sync_state
+-----|----------|----------|------------------|-------------|-------------|-------------------------------|-----------|---------------|----------------|----------------|-----------------|---------------|------------
+  57 |       10 | postgres | ha-postgres-0    | 172.17.0.6  |       52730 | 2018-02-09 04:33:06.051716|00 | streaming | 0/7000060     | 0/7000060      | 0/7000060      | 0/7000060       |             0 | async
+  58 |       10 | postgres | ha-postgres-2    | 172.17.0.8  |       42824 | 2018-02-09 04:33:09.762168|00 | streaming | 0/7000060     | 0/7000060      | 0/7000060      | 0/7000060       |             0 | async
+
+You can see here, now `ha-postgres-0` and `ha-postgres-2` are streaming asynchronously from `ha-postgres-1`, our *primary* server.
+
+<p align="center">
+  <kbd>
+    <img alt="recovered-postgres"  src="/docs/images/postgres/ha-postgres.gif">
+  </kbd>
+</p>
+
+[//]: # (If you want to know how this failover process works, [read here])
+
+## Streaming Replication with `hot standby`
+
+Streaming Replication also works with one or more *hot standby* servers.
+
+```yaml
+apiVersion: kubedb.com/v1alpha1
+kind: Postgres
+metadata:
+  name: hot-postgres
+  namespace: demo
+spec:
+  version: 9.6
+  replicas: 3
+  standby: hot
```

Comment:
Rename to `standbyMode: hot`

Reply:
Only `mode` ? Or `StandbyMode` ? If we add field named `mode`, it will denote Postgres Database mode.

Reply:
I think standbyMode sounds better